Remove (most) Python 2 support from Python code

dreiss opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35625 Remove structseq_slice
* #35614 Remove (most) Python 2 support from C++ code
* #35611 Add errors if we try to use PyTorch with Python 2
* #35627 Remove __future__ imports from many files
* #35624 Remove a use of exec
* #35623 Remove a use of checkScript(str)
* #35622 Remove caffe2/python/compatibility
* #35621 Remove _get_wrapped_func
* #35620 Remove _compatible_subtest
* #35619 Remove _reset_warning_registry
* #35618 Remove _uses_true_division
* #35617 Remove (most) Python 2 support from setup.py
* #35616 Remove (most) Python 2 support from CI scripts
* **#35615 Remove (most) Python 2 support from Python code**

Summary:
Python 2 has reached end-of-life and is no longer supported by PyTorch.
Now we can clean up a lot of cruft that we put in place to support it.
These changes were all done manually, and I skipped anything that seemed
like it would take more than a few seconds, so I think it makes sense to
review it manually as well (though using side-by-side view and ignoring
whitespace change might be helpful).

Test Plan:
CI

Differential Revision: [D20842886](https://our.internmc.facebook.com/intern/diff/D20842886)

dreiss committed:
Remove (most) Python 2 support from Python code

Summary:
Python 2 has reached end-of-life and is no longer supported by PyTorch.
Now we can clean up a lot of cruft that we put in place to support it.
These changes were all done manually, and I skipped anything that seemed
like it would take more than a few seconds, so I think it makes sense to
review it manually as well (though using side-by-side view and ignoring
whitespace change might be helpful).

Test Plan:
CI

ezyang commented on caffe2/python/ideep/conv_op_test.py:
```diff
@@ -158,152 +158,6 @@ def test_depthwise_convolution(self, batch_size, gc, dc):
             print(np.max(np.abs(Y2 - Y0)))
             self.assertTrue(False)
 
-    @unittest.skipIf(sys.version_info.major > 2, "broken in python 3")
```
Probably shouldn't delete this, just mark it as expected broken or something? Or maybe the code is moribund and we don't care anyway... I guess probably that is true.

ezyang commented on test/jit/test_class_type.py:
```diff
@@ -802,8 +801,7 @@ def call():  # noqa: E306
 
-        if not PY2:
-            ops.append(truediv)
+        ops.append(truediv)
```
nit: can be moved to the list (don't do it in this PR if everything else is ok)

ezyang commented on test/run_test.py:
```diff
@@ -17,7 +17,6 @@
 from torch.testing._internal.common_utils import TEST_WITH_ROCM, shell
 import torch.distributed as dist
-PY2 = sys.version_info <= (3,)
 PY33 = sys.version_info >= (3, 3)
```
We can probably axe this one too

ezyang commented on test/test_jit.py:
```diff
@@ -7833,14 +7831,6 @@ def fn(x, y, z):
-        self.assertExpected(str(ast))
-
-    @unittest.skipIf(PY2, "Requires python 3")
     def test_python_frontend_py3(self):
```
nit: should probably rename the test, but whatever

ezyang commented on torch/testing/_internal/common_utils.py:
```diff
@@ -246,14 +238,7 @@ def _check_module_exists(name):
-        except ImportError:
-            return False
-    elif not PY34:  # Python [3, 3.4)
+    if not PY34:  # Python [3, 3.4)
```
Probably can get rid of this one (we don't support 3.4)

dreiss committed:
Update on "Remove (most) Python 2 support from Python code"


Summary:
Python 2 has reached end-of-life and is no longer supported by PyTorch.
Now we can clean up a lot of cruft that we put in place to support it.
These changes were all done manually, and I skipped anything that seemed
like it would take more than a few seconds, so I think it makes sense to
review it manually as well (though using side-by-side view and ignoring
whitespace change might be helpful).

Test Plan:
CI

dreiss committed:
Update on "Remove (most) Python 2 support from Python code"


Summary:
Python 2 has reached end-of-life and is no longer supported by PyTorch.
Now we can clean up a lot of cruft that we put in place to support it.
These changes were all done manually, and I skipped anything that seemed
like it would take more than a few seconds, so I think it makes sense to
review it manually as well (though using side-by-side view and ignoring
whitespace change might be helpful).

Test Plan:
CI

Differential Revision: [D20842886](https://our.internmc.facebook.com/intern/diff/D20842886)

dreiss committed:
Update on "Remove (most) Python 2 support from Python code"


Summary:
Python 2 has reached end-of-life and is no longer supported by PyTorch.
Now we can clean up a lot of cruft that we put in place to support it.
These changes were all done manually, and I skipped anything that seemed
like it would take more than a few seconds, so I think it makes sense to
review it manually as well (though using side-by-side view and ignoring
whitespace change might be helpful).

Test Plan:
CI

Differential Revision: [D20842886](https://our.internmc.facebook.com/intern/diff/D20842886)

dreiss committed:
Update on "Remove (most) Python 2 support from Python code"


Summary:
Python 2 has reached end-of-life and is no longer supported by PyTorch.
Now we can clean up a lot of cruft that we put in place to support it.
These changes were all done manually, and I skipped anything that seemed
like it would take more than a few seconds, so I think it makes sense to
review it manually as well (though using side-by-side view and ignoring
whitespace change might be helpful).

Test Plan:
CI

Differential Revision: [D20842886](https://our.internmc.facebook.com/intern/diff/D20842886)

[ghstack-poisoned]

dreiss committed:
Update on "Remove (most) Python 2 support from Python code"


Summary:
Python 2 has reached end-of-life and is no longer supported by PyTorch.
Now we can clean up a lot of cruft that we put in place to support it.
These changes were all done manually, and I skipped anything that seemed
like it would take more than a few seconds, so I think it makes sense to
review it manually as well (though using side-by-side view and ignoring
whitespace change might be helpful).

Test Plan:
CI

Differential Revision: [D20842886](https://our.internmc.facebook.com/intern/diff/D20842886)

[ghstack-poisoned]

dreiss committed:
Update on "Remove (most) Python 2 support from Python code"


Summary:
Python 2 has reached end-of-life and is no longer supported by PyTorch.
Now we can clean up a lot of cruft that we put in place to support it.
These changes were all done manually, and I skipped anything that seemed
like it would take more than a few seconds, so I think it makes sense to
review it manually as well (though using side-by-side view and ignoring
whitespace change might be helpful).

Test Plan:
CI

Differential Revision: [D20842886](https://our.internmc.facebook.com/intern/diff/D20842886)

[ghstack-poisoned]
